ethereum/jsonrpc/src/test/java/tech/pegasys/pantheon/ethereum/jsonrpc/internal/methods/eea/PermRemoveNodeTest.java

```diff
@@ -0,0 +1,131 @@
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.response.JsonRpcResponse;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.response.JsonRpcSuccessResponse;
+import tech.pegasys.pantheon.ethereum.p2p.api.P2PNetwork;
+import tech.pegasys.pantheon.ethereum.p2p.permissioning.NodeWhitelistController;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class PermRemoveNodeTest {
+
+  private static final boolean SUCCESS_RESULT = true;
+  private PermRemoveNode method;
+  private static final String METHOD_NAME = "perm_removeNodesFromWhitelist";
+
+  private final String enode1 =
+      "enode://6f8a80d14311c39f35f516fa664deaaaa13e85b2f7493f37f6144d86991ec012937307647bd3b9a82abe2974e1407241d54947bbb39763a4cac9f77166ad92a0@192.168.0.10:4567";
+  private final String enode2 =
+      "enode://6f8a80d14311c39f35f516fa664deaaaa13e85b2f7493f37f6144d86991ec012937307647bd3b9a82abe2974e1407241d54947bbb39763a4cac9f77166ad92a0@192.168.0.10:4567";
+  private final String enode3 =
+      "enode://6f8a80d14311c39f35f516fa664deaaaa13e85b2f7493f37f6144d86991ec012937307647bd3b9a82abe2974e1407241d54947bbb39763a4cac9f77166ad92a0@192.168.0.10:4567";
+  private final String badEnode = "enod://dog@cat:fish";
+
+  @Mock private P2PNetwork p2pNetwork;
+  @Mock NodeWhitelistController nodeWhitelistController;
+
+  private JsonRpcParameter params = new JsonRpcParameter();
+
+  @Before
+  public void setUp() {
+    method = new PermRemoveNode(p2pNetwork, params);
+  }
+
+  @Test
+  public void shouldReturnCorrectMethodName() {
+    assertThat(method.getName()).isEqualTo(METHOD_NAME);
+  }
+
+  @Test
+  public void shouldThrowInvalidJsonRpcParametersExceptionWhenBadEnode() {
+    final JsonRpcRequest request = buildRequest(badEnode);
+    final JsonRpcResponse expected =
+        new JsonRpcErrorResponse(request.getId(), JsonRpcError.INVALID_PARAMS);
+    final JsonRpcResponse actual = method.response(request);
+
+    assertThat(actual).isEqualToComparingFieldByFieldRecursively(expected);
+  }
+
+  @Test
+  public void shouldThrowInvalidJsonRpcParametersExceptionWhenNoEnode() {
+    final JsonRpcRequest request = buildRequest("");
+    final JsonRpcResponse expected =
+        new JsonRpcErrorResponse(request.getId(), JsonRpcError.INVALID_PARAMS);
+    final JsonRpcResponse actual = method.response(request);
+
+    assertThat(actual).isEqualToComparingFieldByFieldRecursively(expected);
+  }
+
+  @Test
+  public void shouldThrowExceptionWhenEnodeNotInWhitelist() {
+    final JsonRpcRequest request = buildRequest("");
+    final JsonRpcResponse expected =
+        new JsonRpcErrorResponse(request.getId(), JsonRpcError.INVALID_PARAMS);
+
+    final JsonRpcResponse actual = method.response(request);
+
+    assertThat(actual).isEqualToComparingFieldByFieldRecursively(expected);
+  }
+
```

Comment:
this test looks the same as shouldThrowInvalidJsonRpcParametersExceptionWhenNoEnode - missing buildRequest() param value?

Reply:
Actually, I tested this logic in NodeWhitelistControllerTest.  I'll delete this test.